### PR TITLE
Undo delete

### PR DESCRIPTION
**Pull request by JustKong13**
### Summary 

This pull request is to implement the ability to undo the deletion of a group if a deletion was done by accident. If the trash can on a group was clicked, it would display a button to confirm the deletion of the group, or the ability to undo the delete. 

- [x] implemented functions to undo a delete
- [x] Completion of the front end functionality 

### Test Plan 

image

### CODE REVIEW

File: backend/functions/src/matching/functions.ts [hunk 1]
```diff
@@ -338,6 +338,20 @@ async function hideEmptyGroup(courseId: string, groupNumber: number) {
...
+    .collection('groups')
+    .doc(groupNumber.toString())
+    .update({ hidden: false })
+    .catch((err) => {
+      console.log(err)
+      throw new Error(`Cannot show group ${groupNumber}`)
+    })
+  await Promise.all([groupToHide])
+}
+
```
Comment (mluo24) [inline backend/functions/src/matching/functions.ts]: I think you can change the function to be more general, in that you just pass in `true` or `false` depending on the use case. The endpoint can remain separate, but this function used to implement those endpoints can be more general.

File: backend/functions/src/matching/routes.ts [hunk 2]
```diff
@@ -106,4 +107,14 @@ router.post('/hide-group', (req, res) => {
     })
 })
 
+router.post('/unhide-group', (req, res) => {
+  const { courseId, groupNumber } = req.body
+  unhideGroup(courseId, groupNumber)
```
Comment (mluo24) [inline backend/functions/src/matching/routes.ts]: This line (112) is where you can replace with a more general function, and also in the other endpoint.

**Review by mluo24 — APPROVED** [review]
Thanks for working on this fix Justin! This provides an extra layer of protection if LSC didn't mean to delete a group, which is helpful!

Your current implementation isn't exactly what I had in mind, but there are no designs for this as of now, so I will approve it. I think it is slightly interesting to delete it and then restore it at the confirmation stage, as semantically, "confirming" generally indicates it will not be deleted until you click "confirm." I was more thinking of a popup at the bottom that says "undo" if you want to delete (which is similar to if you delete an event on Google Calendar). 

But besides my comments, everything seems to be working and looks good! Nice work, I will be approving.

**Comment by JustKong13** [comment]
The suggestion to create a more general function that takes in an additional boolean was a good one to clean up the code and not have redundant functions. I've kept the separate endpoints and the two separate calls in `GroupCard.tsx`, but everywhere else should use the same function now.

**Pull request merged**